Text Filter support

moser4035 opened the pull request.

Commit:
Merge branch 'feature/java1_7' into develop

Commit:
Merge branch 'develop' of https://github.com/bitmovin/bitmovin-java into develop

Commit:
Added encoding with watermark filter example

Commit:
Merge branch 'develop' of https://github.com/bitmovin/bitmovin-java into develop

Commit:
Merge branch 'feature/watermark_filter_example' into develop

Commit:
Added text filter support
Added Font enum
Added TextFilter class
Added StreamFilter constructor
Added example for text filter CreateEncodingWithBurnInTextFilter.java

Commit:
Adjusted CreateEncodingWithBurnInTextFilter.java example
Adjusted H264Representation class to store additional information

Commit:
Adjusted CreateEncodingWithWatermarkFilter.java example

msmole commented on src/test/java/com/bitmovin/api/examples/util/H264Representation.java:
```diff
@@ -11,6 +11,9 @@
  */
 public class H264Representation
 {
+    private String outputBasePath;
```
Why do we need that? That should probably not be here.
The H264Representation does not care about inputs nor outputs.

moser4035 replied:
That is a Helper class that is needed to keep track of the generated representations for the manifest generations.

msmole commented on src/test/java/com/bitmovin/api/examples/CreateEncodingWithWatermarkFilter.java:
```diff
@@ -0,0 +1,447 @@
+        final StreamFilter streamFilter = new StreamFilter(watermarkFilter.getId(), 0);
+        StreamFilterList streamFilterList = new StreamFilterList();
+        streamFilterList.setFilters(
+                new ArrayList<StreamFilter>(){{
```
Use `Arrays.asList`?

msmole commented on src/test/java/com/bitmovin/api/examples/CreateEncodingWithBurnInTextFilter.java:
```diff
@@ -0,0 +1,454 @@
+        add(new H264Representation(360, null, null, 800000L, ProfileH264.HIGH, OUTPUT_BASE_PATH, "video/360p"));
+        add(new H264Representation(480, null, null, 1200000L, ProfileH264.HIGH, OUTPUT_BASE_PATH, "video/480p"));
+        add(new H264Representation(720, null, null, 2400000L, ProfileH264.HIGH, OUTPUT_BASE_PATH, "video/720p"));
+        add(new H264Representation(1080, null, null, 4800000L, ProfileH264.HIGH, OUTPUT_BASE_PATH, "video/1080p"));
```
Specifying the width is good! However, you used the height values, e.g., 1080 instead of 1920. Please use the width values here instead of the height values.

msmole commented on src/test/java/com/bitmovin/api/examples/CreateEncodingWithBurnInTextFilter.java:
```diff
@@ -0,0 +1,454 @@
+        HlsManifest manifestHls = this.createHlsManifest("manifest.m3u8", manifestDestination);
+
+        MediaInfo audioMediaInfo = new MediaInfo();
+        audioMediaInfo.setName("audio.m3u8");
```
Name it just 'audio'.

msmole reviewed (changes requested):
Update the TextFilter example to print a timecode at the upper center position of the frame.

Commit:
Adjusted example.
Changed to use Arrays.asList()
Changed to use expressions for setting x of textFilter

Commit:
Changed to use asList()
Adjusted resolutions

Commit:
Adjusted resolutions
Added time code to text filter example

Commit:
Adjusted example

Commit:
Adjusted example

msmole commented on src/test/java/com/bitmovin/api/examples/CreateEncodingWithBurnInTextFilter.java:
```diff
@@ -0,0 +1,451 @@
+    private static String OUTPUT_BASE_PATH = "/your/output/base/path";
+
+    private List<H264Representation> h264Representations = Arrays.asList(
+            new H264Representation(null, 240, null, 400000L , ProfileH264.HIGH, OUTPUT_BASE_PATH, "video/240p"),
```
Please specify the width instead of the height. E.g., 1920 vs 1080.

msmole commented on src/test/java/com/bitmovin/api/examples/CreateEncodingWithWatermarkFilter.java:
```diff
@@ -0,0 +1,444 @@
+    private static String OUTPUT_BASE_PATH = "/your/output/base/path";
+
+    private List<H264Representation> h264Representations = Arrays.asList(
+            new H264Representation(null, 240, null, 400000L , ProfileH264.HIGH, OUTPUT_BASE_PATH, "video/240p"),
```
Please use the width instead of the height, e.g., 1920 vs 1080.

Commit:
Adjusted resolutions of examples
